org.eclipse.xtext.ide/src/org/eclipse/xtext/ide/server/occurrences/DefaultDocumentHighlightService.java

```diff
@@ -187,10 +189,29 @@ protected boolean isDocumentHighlightAvailableFor(final EObject selectedElement,
 			return false;
 		}
 
+		// This code handles the special case where your language has constructs that can refer to
+		// themselves. For example "function MyFunction begin ... end MyFunction" defines the function "MyFunction" and
+		// terminates its implementation block with an additional repetition of the word "MyFunction". Normally, when
+		// you are positioned on a selected element, you only want to highlight that selected element when you are
+		// positioned directly on top of the name of the selected element. However, when the selected element can refer
+		// to itself then there are references inside the element that must trigger highlighting.  In the example,
+		// we also want to highlight "MyFunction" when we are position on the "end MyFunction".
```

Comment:
s/position/positioned/

Reply:
Good catch.  At least someone who appreciates a good block of comments ;)